lib/Data/Validate/Sanctions.pm

```diff
@@ -80,14 +80,25 @@ sub get_sanctioned_info { ## no critic (RequireArgUnpacking)
     }
 
     my $data = $self->_load_data();
+    
+    # Sub to remove non-alphabets from the name
+    my $clean_names = sub {
+        
+        my ($full_name) = @_;
+        
+        # Remove non-alphabets
+        my @cleaned_full_name = map { my $x = $_; $x =~ s/[^[:alpha:]\s]/ /g; $x } split(' ', $full_name);
+        
+        # Remove trailing and leading whitespaces
+        @cleaned_full_name = map { my $x = $_; $x =~ s/^\s*(.*?)\s*$/$1/; split(' ', uc($x)) } @cleaned_full_name;
+
+        return @cleaned_full_name;
+    };
 
-    # prepare list of possible variants of names: LastnameFirstname and FirstnameLastname
-    my @full_name = ($first_name, $last_name || ());
+    my $client_full_name = join(' ', $first_name, $last_name || ());
```

Comment:
Seems that `$client_full_name` isn't used elsewhere. We don't need to join first and last name just to have it split by a sub I think. `clean_names` can just take in an array of names.